src/optimization/duality.jl

```diff
@@ -75,9 +75,15 @@ function activation_value(layer::Layer,
     b_hat = approximate_affine_map(layer, bound)
     l_hat, u_hat = low(b_hat), high(b_hat)
     l, u = layer.activation(l_hat), layer.activation(u_hat)
-
-    o += sum(symbolic_max.(μᵢ .* l_hat, μᵢ .* u_hat))
-    o += sum(symbolic_max.(λᵢ .* l,     λᵢ .* u))
+    for j in 1:length(l)
+        if layer.activation == ReLU()
+            s_max = symbolic_max(μᵢ[j] * l_hat[j] - λᵢ[j] * l[j], μᵢ[j] * u_hat[j] - λᵢ[j] * u[j])
+            l_hat[j] * u_hat[j] >= 0 || @constraint(μᵢ[j].model, s_max >= 0.0)
```

Comment:
Just want to confirm that the logic is `l*mu ≥ 0 OR constraint(s ≥ 0)`

Reply:
Lets also clean up these lines. e.g. 
```julia
if layer.activation == ReLU()
    s_max = @. symbolic_max(μᵢ*l̂ - λᵢ*l, μᵢ*û - λᵢ*u)
    same_sign_inds = findall(l̂ .* û .>= 0)
    @constraint(μᵢ[same_sign_inds].model, s_max[same_sign_inds] .>= 0.0)
else
    s_max = @. symbolic_max(μᵢ * l̂, μᵢ * û) + symbolic_max(-λᵢ * l, -λᵢ * u)
end
```
and using the updated notation from the paper (`g(l)`, `g(u)`,`g(0)`)

Reply:
it seems like the original 

`l_hat[j] * u_hat[j] >= 0 || @constraint(μᵢ[j].model, s_max >= 0.0)`

and the new

`same_sign_inds = findall(l̂ .* û .>= 0)`
    `@constraint(μᵢ[same_sign_inds].model, s_max[same_sign_inds] .>= 0.0)`

Would add constraints in opposite cases. The first would add a constraint when l_hat * u_hat <= 0 is false, while the second would add a constraint when l_hat * u_hat >= 0 is true. I may have flipped something though (I'm still getting used to thinking through the || and && lazy execution).

Reply:
Nice catch